macros/detect_column_changes.sql

```diff
@@ -0,0 +1,41 @@
+{% macro detect_column_changes(a_relation, b_relation, primary_key, exclude_columns=[]) %}
+
+{% set column_names = dbt_utils.get_filtered_columns_in_relation(from=a_relation, except=exclude_columns) %}
+
+with bool_or as (
+
+    select 
+        true as anchor
+        {% for column in column_names %}
+            {% set compare_statement %}
+                (a.{{ column | lower }} != b.{{ column | lower }}
+                or a.{{ column | lower }} is null and b.{{ column | lower }} is not null
+                or a.{{ column | lower }} is not null and b.{{ column | lower }} is null)
+            {% endset %}
+        
+        , {{ dbt.bool_or(compare_statement) }} as {{ column | lower }}_is_changed
```

Comment:
instead of having `true as anchor`, why not 
```suggestion
        {% if not loop.first %}, {% endif %} {{ dbt.bool_or(compare_statement) }} as {{ column | lower }}_is_changed
```

Reply:
because i thought the other way was more fun